### PR TITLE
Linker: fix handling of fragments

### PR DESCRIPTION
**Pull request by calebcordry**
Fixes a regression introduced in https://github.com/ampproject/amphtml/pull/22361

We should not overwrite href if linker param is not added.

Fixes b/134696625

### CODE REVIEW

File: extensions/amp-analytics/0.1/test/test-linker-manager.js [hunk 1]
```diff
@@ -628,6 +628,30 @@ describes.realWin('Linker Manager', {amp: true}, env => {
       });
     });
 
+    it('should not rewrite url if href is fragment', () => {
+      const lm = new LinkerManager(ampdoc, config, /* type */ null, element);
+      return lm.init().then(() => {
```
Comment (alanorozco) [inline extensions/amp-analytics/0.1/test/test-linker-manager.js]: drive-by indentation nit: `async/await` 🚗

```js
it('should not rewrite url if href is fragment', async () => {
  const lm = new LinkerManager(ampdoc, config, /* type */ null, element);
  await lm.init();
	// Using a real anchor el here because of a.href getter will actually
  // return the full url, not just the fragment.
  const a = doc.createElement('a');
  // ...
});
```

File: extensions/amp-analytics/0.1/test/test-linker-manager.js [hunk 2]
```diff
@@ -628,6 +628,30 @@ describes.realWin('Linker Manager', {amp: true}, env => {
...
+        // Using a real anchor el here because of a.href getter will actually
+        // return the full url, not just the fragment.
+        const a = doc.createElement('a');
+        a.href = '#hello';
+        a.hostname = 'amp.source.com';
+
+        const setterSpy = sandbox.spy();
```
Comment (alanorozco) [inline extensions/amp-analytics/0.1/test/test-linker-manager.js]: drive-by brevity nit: 🚗
```js
const get = (obj, prop) => obj[prop];
const set = sandbox.spy();
const aProxy = new Proxy(a, {get, set});
```

File: extensions/amp-analytics/0.1/test/test-linker-manager.js [hunk 3]
```diff
@@ -628,6 +628,30 @@ describes.realWin('Linker Manager', {amp: true}, env => {
...
+          get: (obj, prop) => {
+            return obj[prop];
+          },
+          set: setterSpy,
+        });
+
+        anchorClickHandlers.forEach(handler =>
+          handler(aProxy, {type: 'click'})
+        );
+        expect(setterSpy.notCalled).to.be.true;
```
Comment (alanorozco) [inline extensions/amp-analytics/0.1/test/test-linker-manager.js]: drive-by test convention nit: 🚗
```js
expect(set).not.to.have.been.called;
```
This will give you more specific error messages since it knows it's a function. You can go further to alias it:

```js
expect(set, 'set on a[href=#fragment]').not.to.have.been.called;
```

**Comment by calebcordry** [comment]
Merging this to try to make canary. Will make follow up with TODO to move to navigation.js and nits.
